### ISSUE TITLE
Make config hostgroup id (8000) a configurable option

### ISSUE THREAD

**Issue by markusalbe** [Tusamarco/pxc_scheduler_handler#28]
Customers have HG ids higher than 999 and hence they can't use a configuration HG id of 8000, they would need to use 80000 or some other base value that can be added/subtracted cleanly, so ideally this should be configurable such that users could fit their existing HG ids in the handler way of doing things.

**Comment by Tusamarco** [comment]
Hum... while of course this can be done. I wonder where all the other 998 ids were used for. Does the customer deal with 499 clusters?
Is this not just a side effect of bad configuration?

**Comment by markusalbe** [comment]
They have existing cluster with set id's to 4000+, not sure what's the reason, but customer mentioned they would much prefer to stick to their existing id's.

**Comment by Tusamarco** [comment]
Ok will check how to implement but Low priority

**Comment by Tusamarco** [comment]
Scope
-----------
Make possible to define the value range of the special groups.
At the moment there are two special hostgroups:
	8000 for configuration and
	9000 to isolate nodes when needed without delete them.
The value of the configuration Hostgroups is currently assign with the parameters:
	bckHgW
	bckHgR
	while the HGs for maintenance is static. These should become the same as config HGs parameter of the cluster.


Implementation
-----------------
Two new values will be added to the configuration files:
	- offLineHgW
	- offLineHgR
	Both defaulted to 9000 + HG r/w
	
The values will be assigned to the configurator and assign to the cluster for R/W Hgs.

Any reference to static 8000 or 9000 must be replaced by these dynamic values from PXC cluster object.

**Comment by Tusamarco** [comment]
Implemented FR-28:
2 new parameters added:
- configHgRange
- maintenanceHgRange
2 parameters deprecated:
- bckHgW
- bckHgR

Object PxcCluster/ProxySQL node modified:
- removed all static references
- add dynamic reference

**Comment by Tusamarco** [comment]
all testing are OK going to add another one specifically for this feature.
```
==============================================================================
                  TEST NAME                       RESULT  TIME (ms) COMMENT
------------------------------------------------------------------------------
[  1%] proxysql.reader_add1                      [ pass ]  61030
[  3%] proxysql.reader_add2                      [ pass ]  60495
[  5%] proxysql.reader_add3                      [ pass ]  60919
[  7%] proxysql.reader_add4                      [ pass ]  61386
[  9%] proxysql.reader_desync1                   [ pass ]  61470
[ 11%] proxysql.reader_desync2                   [ pass ]  60942
[ 13%] proxysql.reader_desync3                   [ pass ]  61095
[ 15%] proxysql.reader_desync4                   [ pass ]  61097
[ 17%] proxysql.reader_kill1                     [ pass ]  242608
[ 19%] proxysql.reader_kill2                     [ pass ]  244036
[ 21%] proxysql.reader_kill3                     [ pass ]  242333
[ 23%] proxysql.reader_kill4                     [ pass ]  243314
[ 25%] proxysql.reader_reject_queries1           [ pass ]  60825
[ 27%] proxysql.reader_reject_queries2           [ pass ]  60849
[ 29%] proxysql.reader_reject_queries3           [ pass ]  61463
[ 31%] proxysql.reader_reject_queries4           [ pass ]  61415
[ 33%] proxysql.reader_to_maintenance1           [ pass ]  60849
[ 35%] proxysql.reader_to_maintenance2           [ pass ]  60643
[ 37%] proxysql.reader_to_maintenance3           [ pass ]  60792
[ 39%] proxysql.reader_to_maintenance4           [ pass ]  60938
[ 41%] proxysql.reader_to_readonly1              [ pass ]  60911
[ 43%] proxysql.reader_to_readonly2              [ pass ]  61265
[ 45%] proxysql.reader_to_readonly3              [ pass ]  61448
[ 47%] proxysql.reader_to_readonly4              [ pass ]  61168
[ 49%] proxysql.writer_add1                      [ pass ]  60760
[ 50%] proxysql.writer_add2                      [ pass ]  61211
[ 52%] proxysql.writer_add3                      [ pass ]  60684
[ 54%] proxysql.writer_add4                      [ pass ]  60600
[ 56%] proxysql.writer_desync1                   [ pass ]  61222
[ 58%] proxysql.writer_desync2                   [ pass ]  60825
[ 60%] proxysql.writer_desync3                   [ pass ]  60798
[ 62%] proxysql.writer_desync4                   [ pass ]  61162
[ 64%] proxysql.writer_kill1                     [ pass ]  241584
[ 66%] proxysql.writer_kill1_persist             [ pass ]  242475
[ 68%] proxysql.writer_kill2                     [ pass ]  242271
[ 70%] proxysql.writer_kill2b                    [ pass ]  242702
[ 72%] proxysql.writer_kill3                     [ pass ]  242077
[ 74%] proxysql.writer_kill4                     [ pass ]  242300
[ 76%] proxysql.writer_reject_queries1           [ pass ]  61477
[ 78%] proxysql.writer_reject_queries2           [ pass ]  60825
[ 80%] proxysql.writer_reject_queries3           [ pass ]  61219
[ 82%] proxysql.writer_reject_queries4           [ pass ]  61284
[ 84%] proxysql.writer_to_maintenance1           [ pass ]  60819
[ 86%] proxysql.writer_to_maintenance2           [ pass ]  61034
[ 88%] proxysql.writer_to_maintenance3           [ pass ]  60936
[ 90%] proxysql.writer_to_maintenance4           [ pass ]  61067
[ 92%] proxysql.writer_to_readonly1              [ pass ]  60760
[ 94%] proxysql.writer_to_readonly2              [ pass ]  61192
[ 96%] proxysql.writer_to_readonly3              [ pass ]  60997
[ 98%] proxysql.writer_to_readonly4              [ pass ]  60903
[100%] shutdown_report                           [ pass ]       
------------------------------------------------------------------------------
The servers were restarted 0 times
The servers were reinitialized 0 times
Spent 4866.475 of 5062 seconds executing testcases

Completed: All 51 tests were successful.
```

**Comment by Tusamarco** [comment]
@markusalbe 
This should be fine. Can you please test it as well?
I am going to merge to main soon but would like to have a thumb up.
```
==============================================================================
                  TEST NAME                       RESULT  TIME (ms) COMMENT
------------------------------------------------------------------------------
[ 50%] proxysql.reader_add1_notdef_conf_hg       [ pass ]  61289
[100%] shutdown_report                           [ pass ]       
------------------------------------------------------------------------------
The servers were restarted 0 times
The servers were reinitialized 0 times
Spent 61.289 of 164 seconds executing testcases

Completed: All 2 tests were successful.

```

**Comment by Tusamarco** [comment]
included in release 1.2.0
